### DISCUSSION TITLE
Notification settings improvement

### DISCUSSION THREAD

**Discussion by Hilyxx** [linuxmint/discussions#489] [Ideas]
Today, notifications are an integral part of the working environment. Giving users control over these notifications is essential.

Cinnamon is a desktop environment with rich features but notification settings has a fairly limited range of options.
Gnome shell, xfce, bungie, KDE all have a notification center that allows the user to enable or disable notifications for each application (without using Dconf editor).

An update of notification settings would be great.

Here's my mock-up:

![notif1](https://github.com/linuxmint/discussions/assets/21991448/98e79e65-ad17-4c08-a28e-c6312eedfee2)
![notif2](https://github.com/linuxmint/discussions/assets/21991448/fc73e553-d7de-4ca4-a1cb-f395f92d81b5)

And thank you LM team for your great work

**Comment by GCAT01** [comment]
I would love to more options for notifications. An option to customize sounds by application would be nice too.

**Comment by Pintjebier** [comment]


**Reply by igorsantos07** [reply]
That's covered by the inhibit applet, so far, and is also a dupe from https://github.com/orgs/linuxmint/discussions/417

**Comment by Jopp-gh** [comment]
as much as I like notifications, the ability to check notifications for each and every app seems a bit excessive for me. No critique to the original post, just my 2 cents. Also Mac allows this notification tuning. 

One thing that bothers me, is the height of notifications, example album covers.  Notifications should use a fixed height. A notification task is to inform you briefly. Notifications are not meant to be "some document" to read.

**Reply by igorsantos07** [reply]
I agree that notification popups should be configurable, as so far they seem way too big (and the X button is sometimes hard to hit).
You should open a separate topic, though, as this one seems focused on configuring apps - it's easier to track the different topics if they're separate :)

**Comment by igorsantos07** [comment]
Configuring which applications can post notifications would be awesome.

I came here looking for a mostly similar feature: disabling which notifications should be stored in the panel's history! I use a screenshot application (flameshot) that posts notifications, which is useful, but there's no reason at all to keep them in the history as I'll never miss one, given they show based on my current interactions. I can see other applications which do not need to be kept in the unread history as well. That panel should have two columns: "allow notifications" and "allow on unread history". This could even be combined to "silence" specific notifications without completely disappearing with them.

**Comment by Secret-chest** [comment]
Also borrow per-app ringtones from Android.

**Comment by OnlineLearningTutorials** [comment]
In notification, I want full screen option improvement. Like when I am working on some app in full screen mode than I can choose whether the notification is to be shown or not. So if I work on some professional work than it should notify even in full screen that some message is present. But if I do some personal work than I can disable the notification in full screen. The gnome has the extension which do the work better. For linuxmint I searched the web, after some js file change the notification shown on fullscreen. But I want it to be in settings to choose whether the notification to be shown or not in full screen mode.

**Reply by Jopp-gh** [reply]
usually notifications are clickable and should open files as well. 

In case you get important news, a simple click on notifications is all you need, to interact consciously with notifications.  

Why?
- There's no need for full screen notifications as it would disrupt work in progress. 
- it would be bad design to force users to react to the next thing popping up on-screen.

**Reply by OnlineLearningTutorials** [reply]
yes, generally it is bad to show notification in fullscreen. but imagine you and your friend talking on some app which shows notification (when not focused) that some new message is present. So it is not necessary that your friend would text you in any 5 minute, he can text you any time or may be not. Now you decided to watch some video like say 1 hour. You check message for initial 10 minute that if your friend not text you. after that you think your friend is busy with other task and watching your remaining 1 hour video. but when you finish the video you find that your friend ask some information like 30 minutes ago, he waited for 10 minutes and now he is working it on next day. You think that if you get some notification alert whether is sound alert or shown whole notification in fullscreen, than you can answer your friend question immediately and not to wait till next day. 

This is some example situation. So I called it would be optional those who not want can disable it and those who want can enable it. It is not necessary to show full notification it may be sound alert that some new notification is present.

It may also be important like you are working on laptop last five hour with battery, battery is critically low and you are reading pdf in full screen mode. you relise battery low when laptop suddently shut down. Not to disturb in full screen is a good idea but not at all situations. That's why I am not want it to compulsory rather in optionally user friendly.

![fullscreen notification](https://github.com/user-attachments/assets/552ba8d9-d668-4efa-b3f3-d239b259e392)

there is a workaround in cinnamon to show critical notification in fullscreen while not the other. I want let the user can select whether to show notification in fullscreen or not rather than hardcoded in messageTray.js. Whether may be main settings or by some extensions (not applet or desklet because there is nothing to be shown its a functionality).

On Ubuntu, when I play long video and talking to someone than I installed the extension [Fullscreen Notifications](https://extensions.gnome.org/extension/1610/fullscreen-notifications/)

**Comment by Jopp-gh** [comment]
I'm no more up to date with other distros than Linux Mint. 
For completeness I want to add, there's a core applet called notifications@cinnamon.org (on screen a bell icon) where you can check all recent notifications.  

Full screen notifications might be a idea for a custom applet, imo.

**Reply by rcalixte** [reply]
There is an open [pull request](https://github.com/linuxmint/cinnamon/pull/11844) for this. The integration runs too deep to simply be an applet.

**Reply by OnlineLearningTutorials** [reply]
exact and simple solution

![Screenshot from 2024-08-25 10-21-36](https://github.com/user-attachments/assets/fef1c49e-f6e2-457c-9705-a039ec6c8e47)
![Screenshot from 2024-08-25 10-25-36](https://github.com/user-attachments/assets/fbf06a88-ec23-494c-b53d-97a352f1b9a0)

writing extension may be solution but this is the simple and best solution.

**Reply by Jopp-gh** [reply]
Nice, if it's a toggle.

Sometimes I get just too many not important–notifications so my skepticism. 

Being able to set notifications for each and every app seems the only way to get "important notifications". Pretty rough to deal with priority notifications, imo. But even Mac has not a "better" system to deal with notifications. IMO, their design ideas are usually well thought out. 

On Mac you can delay a notification in minutes or days, if the app sending the notification added this delay. 

Maybe urgent notifications could be tagged as such, if you accept them immediately versus if you delay them.

So the next time when a notification would pop up your system could guess the urgency and add a background color red or yellow or blue or default colour to notifications.

- Urgency 0 = you don't care or dismiss
- Urgency 1 = delay in minutes
- Urgency 2 = delay in days
- Urgency 3 = you click on the notification to open an app or document requiring your attention

Even this method is rough and needs some more details to work nicely.

**Comment by trapexit** [comment]
It'd be nice if the notifications actually could be turned off. If a notification comes in after toggling them off they still pop up and if I try to run another notification service it reports that cinnamon is org.freedesktop.Notifications on dbus. If the notifications won't be improved at least allow using a different notification package.

**Reply by cplittlfield** [reply]
> It'd be nice if the notifications actually could be turned off. 

You mean, as the Cinnamon settings already provide for?

**Comment by Crispycjr** [comment]
My 2 cents on notifications just involves a personal gripe with how it handles with Spotify. Even with 'Ignore transient notifications' turned on, the bell remains open even when 'show empty tray' is ticked on. I'd perhaps like to see more application-specific settings regarding this particular quirk.

This could be an app-specific issue, but I figured I might as well mention it.

**Comment by ghost** [comment]
Good idea, I approve. 

Also an additional idea to this: make it possible to fine-tune positioning of the notifications (not just top-right or bottom-right)
For example use case: I moved my panel to the top, and moved the clock to the middle, I'd love to have my notifications pop up in the middle of my screen at the top, so it would feel a bit more clean and professional.

Borrowing a sample image from Plasma, so the devs will have a better understanding:
<details> <summary> Sample </summary>

![image](https://github.com/user-attachments/assets/f081b419-895c-47cc-a99d-6a2c0c5bf0dd)
</details>

**Comment by Crispycjr** [comment]
One more thing, less related to the settings and more toward design in case someone sees this. I think perhaps the glowing outline around the notification is tacky, and unnecessary.

<img width="529" height="346" alt="491910800-fce1a8a4-6dbd-4142-b833-cfe72f35626e" src="https://github.com/user-attachments/assets/0e3dc477-22f5-4c78-ab30-d1dcbaf89b39" />

<img width="308" height="140" alt="Untitled" src="https://github.com/user-attachments/assets/f8536338-1f09-4d78-b943-e09f64bb26ad" />



It doesn't look good against darker backgrounds, and the blue stripe is already enough to signify itself. It does make sense that it helps contrast the window against the background in the notification applet, but I feel like the outline made by the volume/workspace OSD would be enough to replace it as far as design goes.

**Reply by cplittlfield** [reply]
When making such design decisions, please bear in mind those of us who are older, and whose eyesight is not what it once was.  One of the big problems I have with the current design ethos is coping with messages that are the same color as the main app and that have no border to distinguish them from the background.  In such cases, it is very hard for these old eyes to register that a message has been displayed.

Making messages stand out might be considered tacky in terms of current design thinking, but it is extremely helpful for those of us whose vision is impaired.  Please bear us in mind when you are designing.

**Reply by Secret-chest** [reply]
Visual design can be done in CSS themes for Cinnamon, it is already possible.

**Reply by Jopp-gh** [reply]
Css is actually very easy to code.

Mint shouldn't "modernize" and force aesthetics on the users at all, (what started with Xia and continues till today) **because we have themes for that**.

Mint should deliver instead, build-in tools to allow users to create custom css themes easily and out of the box. 

We need tools to display css parameters interactively so any changes are visible immediately. This is the way to build themes.

**Comment by Mosblinker** [comment]
Please implement this. I've had to disable all notifications since there's no way of customizing the way notifications work and I was getting tired of getting a popup every time someone sends me a message on Discord. Also, as far as I know, there's no Do Not Disturb mode so I can't turn off notifications for the night to pause notifications when I'm trying to sleep.

**Comment by igorsantos07** [comment]
@Hilyxx it looks like we're having great news lately! I guess the things you requested are getting implemented in the next Mint version? Would you mind sharing with everyone on the thread more details as to why you closed the discussion as _outdated_?

**Comment by lntegrity** [comment]
Here 6 months later ^ to say I hope the LM team is still possibly tracking this even though Hilyxx marked it as closed/resolved (with no actual resolution marked that I can see, anyway).  I would love to see that mock-up implemented just the way it's presented - toggles for each app that has the ability to call system notifications 👍🏻

**Reply by igorsantos07** [reply]
@Hilyxx

**Comment by mtwebster** [comment]
https://github.com/linuxmint/cinnamon/pull/13598

**Reply by lntegrity** [reply]
I found *this* item (489) as a duplicate of https://github.com/orgs/linuxmint/discussions/1125 (which a search engine had at least crawled), and didn't think about the existence of yet-other duplicates.  Thank you for linking this 👍

**Comment by Hilyxx** [comment]
https://github.com/linuxmint/cinnamon/pull/13598

**Reply by igorsantos07** [reply]
YASSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
